test-suite/src/test/scala/org/scalajs/testsuite/jsinterop/TupleTest.scala

```diff
@@ -0,0 +1,42 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.jsinterop
+
+import org.scalajs.jasminetest.JasmineTest
+
+import scala.scalajs.js
+
+object TupleTest extends JasmineTest {
+
+  describe("scala.scalajs.js.Tuple") {
+
+    it("should provide an equivalent of scala tuple") {
+      val obj: js.Tuple2[Int, String] = (42, "foobar")
+
+      expect(obj._1).toEqual(42)
+      expect(obj._2).toEqual("foobar")
+    }
+
+    it("should convert to scala tuple") {
+      val obj: js.Tuple2[Int, String] = (42, "foobar")
+      val scalaObj: scala.Tuple2[Int, String] = obj
+
+      expect(scalaObj._1).toEqual(42)
+      expect(scalaObj._2).toEqual("foobar")
+    }
+
+    it("should unapply js tuple") {
+      val obj: js.Tuple2[Int, String] = (42, "foobar")
+      val js.Tuple2(t1, t2) = obj
```

Comment:
You should somehow test that `t1` and `t2` can be inferred as `Int` and `String`, e.g., by adding this afterwards:
```scala
val t1IsInt: Int = t1
val t2IsString: String = t2
```